crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -75,9 +74,11 @@ where
     precompile_cache_disabled: bool,
     /// Precompile cache map.
     precompile_cache_map: PrecompileCacheMap<SpecFor<Evm>>,
-    /// A cleared accounts sparse trie, kept around to be reused for the state root computation so
+    /// A cleared `SparseStateTrie`, kept around to be reused for the state root computation so
     /// that allocations can be minimized.
-    accounts_trie: Arc<Mutex<Option<SparseTrie<ConfiguredSparseTrie>>>>,
+    sparse_state_trie: Arc<
+        tokio::sync::Mutex<Option<ClearedSparseStateTrie<ConfiguredSparseTrie, SerialSparseTrie>>>,
```

Comment:
Is there a specific reason for using tokio mutex here, afaict we only hold the lock guard across memory operations

what this does is basically `future::block_on(mutex.lock())`, 
a parkinglot mutex would be more appropriate here, since we never use this in async context